save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/OrganizationController.kt

```diff
@@ -99,20 +135,43 @@ internal class OrganizationController(
     }
 
     /**
-     * @param organization updateOrganization
+     * @param organizationName name of an organization that should be changed
+     * @param organization draft organization that should be saved as an organization with name [organizationName]
      * @param authentication an [Authentication] representing an authenticated request
      * @return response
      */
     @PostMapping("/{organizationName}/update")
-    @Suppress("UnsafeCallOnNullableType")
-    fun updateOrganization(@RequestBody organization: Organization, authentication: Authentication): Mono<StringResponse> = Mono
-        .just(organization)
-        .filter { organizationPermissionEvaluator.hasGlobalRoleOrOrganizationRole(authentication, it.name, Role.OWNER) }
-        .map {
-            organizationService.updateOrganization(organization)
+    fun updateOrganization(
+        @PathVariable organizationName: String,
+        @RequestBody organization: Organization,
+        authentication: Authentication,
+    ): Mono<StringResponse> = Mono.just(
+        organizationName
+    )
+        .flatMap {
+            organizationService.findByName(it).toMono()
```

Comment:
In OrganizationView in componentDidMount we get the _organization_ from the database. We change something in it and return it to the backend. The organization already contains an ID. We don't have to look it up again in the database by name.
The _save(Organization)_ method will not create a new entry, but will update the old one by ID

Reply:
We generally can't rely on *all* fields of an object received from a network request. For example, to check `canCreateContests` for a certain organization we have to look it up in the database by name or by ID, because user may have sent a request with maliciously altered value. Same goes for the combination of name and ID. So we still need to use some key to look up the real data for an entity and operate with it.

Reply:
Ok. Can we then create an OrganizationDTO entity to pass data between the frontend and backend?

Reply:
I guess there is an issue for that

Reply:
There is a similar one https://github.com/saveourtool/save-cloud/issues/363

Reply:
Yes. I forgot. OK. Let's leave it as is for now. But we will have to think about creating this layer in the future for the project and organization in the future.